src/main/java/seedu/address/logic/commands/findcommands/FindPersonCommand.java

```diff
@@ -0,0 +1,49 @@
+package seedu.address.logic.commands.findcommands;
+
+import static java.util.Objects.requireNonNull;
+
+import seedu.address.logic.Messages;
+import seedu.address.logic.commands.CommandResult;
+import seedu.address.model.Model;
+import seedu.address.model.person.NameAndTagContainKeywordsPredicate;
+
+/**
+ * Finds and lists all persons in address book whose name contains any of the argument keywords in their names or tags.
+ * Keyword matching is case-insensitive.
+ */
+public class FindPersonCommand extends FindCommand {
+    public static final String MESSAGE_SUCCESS = "Success message for the find command!";
+
+    private final NameAndTagContainKeywordsPredicate predicate;
+
+    /**
+     * Creates a FindPersonCommand to find persons with the specified {@code NameAndTagContainKeywordsPredicate}
+     */
+    public FindPersonCommand(NameAndTagContainKeywordsPredicate predicate) {
+        assert predicate != null;
```

Comment:
This is okay but why not use requireNonNull?

Reply:
I will change it to requireNonNull but honestly I don't know what is the difference haha

Reply:
Same, I think it does the same thing haha

better abstraction maybe